sympy/core/relational.py

```diff
@@ -170,7 +170,44 @@ def equals(self, other, failing_expression=False):
                     return r
                 return l
 
+    def _simplify_relational(self):
```

Comment:
I am not sure about this function , but you can try something like if `lhs != rhs` and
` (lhs-rhs)` has not any `symbols` then return `S.false`. 
around [this line](https://github.com/sympy/sympy/blob/master/sympy/core/relational.py#L290).
This will not affect during simplification also is